app/components/Texting/Texting.jsx

```diff
@@ -0,0 +1,223 @@
+import React, { useState } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import './Texting.scss';
+import { images } from 'assets';
+import * as dataService from '../../utils/DataService';
+import Loader from '../ui/Loader';
+
+// Initial state's values
+const initialState = {
+  userName: '',
+  phoneNumber: '',
+  agreed: false,
+  isLoading: false,
+  isSent: false,
+  error: false,
+};
+
+// Text resource informations to the user phone
+
+const Texting = ({ toggle, resource }) => {
+  const [state, setState] = useState(initialState);
+
+  const {
+    userName,
+    phoneNumber,
+    agreed,
+    isLoading,
+    error,
+    isSent,
+  } = state;
+
+  const {
+    categories,
+    id,
+    addresses,
+    phones,
+  } = resource;
+  // Data to send to the Api, The logic in the values handles the missing infos.
+  const data = {
+    firstName: userName ? userName.match(/\S+/g)[0] : '',
+    lastName: userName ? userName.match(/\S+/g).slice(1).join(' ') : '',
+    mobilePhone: phoneNumber.replace(/\s/g, ''),
+    phoneAlternate: '',
+    phoneHome: '',
+    phoneWork: '',
+    tags: categories,
+    resourceId: id,
+    engagementType: 'Resource Info',
+    engagementInfo: {
+      Org_Name: resource.name,
+      Org_Address1: addresses.length ? addresses[0].address_1 : '',
+      Org_Address2: addresses.length ? addresses[0].address_2 : '',
+      City: addresses.length ? addresses[0].city : '',
+      State: addresses.length ? addresses[0].state_province : '',
+      Zip: addresses.length ? addresses[0].postal_code : '',
+      Org_Phone: phones ? phones[0].number : '',
```

Comment:
Could we use `snake_case` for all of these parameters? The rest of our APIs follow that naming convention, so I think it'd be better to stick with that.

I think you might be passing these directly through to the Textellent API, but if that's the case, you probably should apply the data transformation on our own backend API so that we decouple our frontend code from the Textellent API.

Also, for many of these fields, you can supply the data entirely from our backend API. You can even set the blank default values on some of these fields directly from our backend API so that the frontend code doesn't have to deal with it.